Library/Homebrew/formula.rb

```diff
@@ -1582,6 +1582,39 @@ def time
     end
   end
 
+  # Replaces a universal binary with its native slice.
+  #
+  # If called with no parameters, does this with all compatible
+  # universal binaries in a {Formula}'s {Keg}.
+  sig { params(targets: T.nilable(T.any(Pathname, String))).void }
+  def deuniversalize_machos(*targets)
+    if targets.blank?
+      targets = any_installed_keg.mach_o_files.select do |file|
+        file.arch == :universal && file.archs.include?(Hardware::CPU.arch)
+      end
+    end
+
+    targets.each { |t| extract_macho_slice_from(Pathname.new(t), Hardware::CPU.arch) }
+  end
+
+  # @private
+  sig { params(file: Pathname, arch: T.nilable(Symbol)).void }
+  def extract_macho_slice_from(file, arch = Hardware::CPU.arch)
+    odebug "Extracting #{arch} slice from #{file}"
+    file.ensure_writable do
+      macho = MachO::FatFile.new(file)
+      native_slice = macho.extract(Hardware::CPU.arch)
+      native_slice.write file
+      MachO.codesign! file if Hardware::CPU.arm?
```

Comment:
[`MachO.codesign!`](https://github.com/Homebrew/ruby-macho/blob/bb4b3a393d4ea4adaa49396c77a2ee8aa51dcc67/lib/macho.rb#L51-L60) is slightly different from our [`Keg#apply_ad_hoc_signature`](https://github.com/Homebrew/brew/blob/d45832bbfc7a7e89a7f54cc314c93a087decf89f/Library/Homebrew/os/mac/keg.rb#L64).

Not sure if we have a preference between one or the other. I suspect we might want to just make `MachO.codesign!` more like `Keg#apply_ad_hoc_signature` and use `MachO.codesign!` in `Keg` too.

Reply:
No preference but feels like consistency might be nice (not-blocking to this PR)